quickwit-janitor/src/actors/garbage_collector.rs

```diff
@@ -115,30 +110,71 @@ impl Handler<Loop> for GarbageCollector {
         info!("garbage-collect-operation");
         self.counters.num_passes += 1;
 
-        let deleted_file_entries = run_garbage_collect(
-            &self.pipeline_id.index_id,
-            self.split_store.clone(),
-            self.metastore.clone(),
-            STAGED_GRACE_PERIOD,
-            DELETION_GRACE_PERIOD,
-            false,
-            Some(ctx),
-        )
-        .await?;
-
-        if !deleted_file_entries.is_empty() {
-            let deleted_files: HashSet<&str> = deleted_file_entries
-                .iter()
-                .map(|deleted_entry| deleted_entry.file_name.as_str())
-                .collect();
-            info!(deleted_files=?deleted_files, "gc-delete");
-
-            self.counters.num_deleted_files += deleted_file_entries.len();
-            self.counters.num_deleted_bytes += deleted_file_entries
-                .iter()
-                .map(|entry| entry.file_size_in_bytes as usize)
-                .sum::<usize>();
+        let index_metadatas = self
```

Comment:
We could add a method on the metastore to directly get the splits to delete. I would not do it in this PR but rather open an issue to avoid looping all indexes.

Reply:
actually, we don't really want that as we need to support the file backed metastore :/

Reply:
There is still the need to fetch the index_metadatas since we need to resolve the storage for each index.

Reply:
Indeed, I always forget about it. So we could directly have a method on the metastore to get the list of splits to delete whatever the index.